lib/bolt/pal.rb

```diff
@@ -215,6 +215,7 @@ def in_bolt_compiler
     def with_bolt_executor(executor, inventory, pdb_client = nil, applicator = nil, &block)
       setup
       opts = {
+        bolt_project: @project,
```

Comment:
I think similar to line 176 here we'll want to call `detect_project_conflict` after overriding this.

Reply:
It looks like `detect_project_conflict` will already be called whether using `bolt apply` or an apply block. I'm also not sure how to squeeze that function call into the block that's being passed to `Puppet.override` on line 237. :|

Reply:
Oh I see - we call `in_plan_compiler` to call the `apply_prep` function in cli.rb, which raises the project conflict error :+1: